ext/standard/config.m4

```diff
@@ -51,14 +51,23 @@ if test "$ac_cv_flush_io" = "yes"; then
   AC_DEFINE(HAVE_FLUSHIO, 1, [Define if flush should be called explicitly after a buffered io.])
 fi
 
-PHP_CHECK_FUNC(crypt, crypt)
-PHP_CHECK_FUNC(crypt_r, crypt)
-if test "$ac_cv_func_crypt_r" = "yes"; then
-  PHP_CRYPT_R_STYLE
-fi
-
-AC_CACHE_CHECK(for standard DES crypt, ac_cv_crypt_des,[
-  AC_RUN_IFELSE([AC_LANG_SOURCE([[
+PHP_ARG_WITH([external-libcrypt],
+  [for external libcrypt or libxcrypt],
+  [AS_HELP_STRING([--with-external-libcrypt],
+    [Use external libcrypt or libxcrypt])],
+  [no],
+  [no])
+
+if test "$PHP_EXTERNAL_LIBCRYPT" != "no"; then
+  PHP_CHECK_FUNC(crypt, crypt)
+  PHP_CHECK_FUNC(crypt_r, crypt)
+  AC_CHECK_HEADERS(crypt.h)
```

Comment:
As this check is now here (which I agree is the right place), we should drop the redundant one from configure.ac.